### ISSUE TITLE
Hitting escape after changing value assigns it

### ISSUE THREAD

**Issue by phazei** [dangrossman/daterangepicker#1889]
This is related to 
https://github.com/dangrossman/daterangepicker/issues/1156

If the range is changed on the calendar, and then instead of clicking apply, escape is pressed, just like with the issue of outsideClick, hide() is called and the value is assigned.

The other issue was fixable by the added (& undocumented) outsideClick callback function where clickCancel could be called, but there's no way to not do that with the escape callback.  Maybe a flag could be added to set values back to this.oldStartDate / this.oldEndDate in some cases?

**Comment by dimitrisscript** [comment]
did you try setting

`autoUpdateInput: false`

in your options?

**Comment by phazei** [comment]
Yes, it is always set to false in my implementation.  I'm using https://github.com/fragaria/angular-daterangepicker which I mostly re-wrote for it's latest version.

**Comment by tmbv93** [comment]
Regarding this, but also issue #1156, I think it would be really useful to have an event called outsideClick.daterangepicker or perhaps dimiss.daterangepicker, where the daterangepicker is closed, but neither apply or cancel has been clicked. 

With an event like that, you'd have full control over your app's behavior in cases where the user didn't explicitly intend to apply or clear.
